transaction/tests/DirectedGraph_unittest.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "transaction/DirectedGraph.hpp"
+
+#include <vector>
+
+#include "transaction/Transaction.hpp"
+
+#include "gtest/gtest.h"
+
+namespace quickstep {
+
+namespace transaction {
+
+TEST(DirectedGraphTest, AddNode) {
+  DirectedGraph wait_for_graph;
+  transaction_id *tid3 = new transaction_id(3);
+  transaction_id *tid4 = new transaction_id(4);
+  transaction_id *tid5 = new transaction_id(5);
+  transaction_id *tid6 = new transaction_id(6);
+
+  EXPECT_EQ(0u, wait_for_graph.getNumNodes());
+
+  wait_for_graph.addNodeUnchecked(tid3);
+
+  EXPECT_EQ(1u, wait_for_graph.getNumNodes());
+
+  wait_for_graph.addNodeUnchecked(tid4);
+
+  EXPECT_EQ(2u, wait_for_graph.getNumNodes());
+
+  wait_for_graph.addNodeUnchecked(tid5);
+  wait_for_graph.addNodeUnchecked(tid6);
+
+  EXPECT_EQ(4u, wait_for_graph.getNumNodes());
+}
+
+TEST(DirectedGraphTest, AddEdge) {
+  DirectedGraph wait_for_graph;
+  transaction_id *tid3 = new transaction_id(3);
+  transaction_id *tid4 = new transaction_id(4);
+  transaction_id *tid5 = new transaction_id(5);
+  transaction_id *tid6 = new transaction_id(6);
+
+  using NID = DirectedGraph::node_id;
+  NID nid3 = wait_for_graph.addNodeUnchecked(tid3);
+  NID nid6 = wait_for_graph.addNodeUnchecked(tid6);
+  NID nid4 = wait_for_graph.addNodeUnchecked(tid4);
+  NID nid5 = wait_for_graph.addNodeUnchecked(tid5);
+
+  wait_for_graph.addEdgeUnchecked(nid3, nid5);
+  wait_for_graph.addEdgeUnchecked(nid6, nid4);
+  wait_for_graph.addEdgeUnchecked(nid3, nid6);
+  wait_for_graph.addEdgeUnchecked(nid4, nid6);
+
+  EXPECT_TRUE(wait_for_graph.hasEdge(nid3, nid5));
+  EXPECT_TRUE(wait_for_graph.hasEdge(nid6, nid4));
+  EXPECT_TRUE(wait_for_graph.hasEdge(nid4, nid6));
+  EXPECT_TRUE(wait_for_graph.hasEdge(nid3, nid6));
+
+  EXPECT_FALSE(wait_for_graph.hasEdge(nid5, nid3));
+  EXPECT_FALSE(wait_for_graph.hasEdge(nid6, nid3));
+  EXPECT_FALSE(wait_for_graph.hasEdge(nid4, nid5));
+}
+
+TEST(DirectedGraphTest, GetAdjacentNodes) {
+  DirectedGraph wait_for_graph;
+  transaction_id *tid3 = new transaction_id(3);
+  transaction_id *tid4 = new transaction_id(4);
+  transaction_id *tid5 = new transaction_id(5);
+  transaction_id *tid6 = new transaction_id(6);
```

Comment:
The above five lines are getting reused in all the test fixtures. You can create a ``SetUp()`` method and move these lines from all the fixtures to the ``SetUp()`` method and then switch from ``TEST`` to ``TEST_F``